src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -553,10 +553,10 @@ protected internal void OnDocumentClosed(DocumentId documentId, TextLoader reloa
 
         protected internal void OnAdditionalDocumentClosed(DocumentId documentId, TextLoader reloader)
         {
-            this.CheckAdditionalDocumentIsInCurrentSolution(documentId);
-
             using (_serializationLock.DisposableWait())
             {
```

Comment:
Just curious, why not calling `this.CheckDocumentIsOpen(documentId)` here?

Reply:
Ah, probably because until @mavasani's change that wouldn't have worked. I shall do that.